### ISSUE TITLE
[py2py3] RuntimeError with unsupported type in Configuration.py

### ISSUE THREAD

**Issue by amaltaro** [dmwm/WMCore#10515]
**Impact of the bug**
WMCore

**Describe the bug**
The Configuration module seems not to like the `bytes` complex data type in python3 unit tests, see full traceback in [1]
I believe there is some planned python3 work to be done in the Configuration module, which might be exactly what is reported in this issue.

**How to reproduce it**
```
nosetests test/python/WMCore_t/WMSpec_t/WMWorkload_t.py
````

**Expected behavior**
The Configuration module should be able to deal with strings in a consistent way between python2 and python3, including this possible byte string vs unicode string.

**Additional context and error message**
[1]
```
======================================================================
ERROR: _testBlockCloseSettings_
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/dmwm/wmcore_unittest/WMCore/test/python/WMCore_t/WMSpec_t/WMWorkload_t.py", line 2031, in testBlockCloseSettings
    testWorkload = self.makeTestWorkload()[0]
  File "/home/dmwm/wmcore_unittest/WMCore/test/python/WMCore_t/WMSpec_t/WMWorkload_t.py", line 136, in makeTestWorkload
    harvestTaskCMSSWHelper.setDataProcessingConfig("pp", "dqmHarvesting",
  File "/home/dmwm/wmcore_unittest/WMCore/src/python/WMCore/WMSpec/Steps/Templates/CMSSW.py", line 159, in setDataProcessingConfig
    self.data.application.configuration.pickledarguments = pickle.dumps(args)
  File "/home/dmwm/wmcore_unittest/WMCore/src/python/WMCore/Configuration.py", line 146, in __setattr__
    self._complexTypeCheck(name, value)
  File "/home/dmwm/wmcore_unittest/WMCore/src/python/WMCore/Configuration.py", line 124, in _complexTypeCheck
    raise RuntimeError(msg)
RuntimeError: Not supported type in sequence:<class 'bytes'>
for name: pickledarguments and value: b'\x80\x04\x95X\x00\x00\x00\x00\x00\x00\x00}\x94(\x8c\tglobalTag\x94\x8c\x05Bogus\x94\x8c\x0bdatasetName\x94\x8c /bogusPrimary/bogusProcessed/DQM\x94\x8c\trunNumber\x94K\x00u.'
Added to WMAgent Configuration.Use ConfigurationEx to skip checks on config params
```

**Comment by amaltaro** [comment]
@mapellidario Dario, if this is exactly what you have been working on lately, then you might be the best person to work on this. Otherwise, I'd assign it to @goughes to make sure he also gets some of these ;)

**Comment by mapellidario** [comment]
I would prefer to take this, since I already have #9974 opened a long time ago and seen some ways that Configuration can silently fail on us. 

However, this is likely going to require some team-wide discussion.

**Comment by mapellidario** [comment]
Actually, I rebased #9974 on top of master and it fixes this issue!

**Comment by mapellidario** [comment]
Sicen we merged #9974, we may need to close this one, too. On dmwm/WMCore - master the unittest

```bash
nosetests test/python/WMCore_t/WMSpec_t/WMWorkload_t.py:WMWorkloadTest.testBlockCloseSettings
```

is not failing for me. I still have 3 unittests failing in `nosetests test/python/WMCore_t/WMSpec_t/WMWorkload_t.py`, but i think that they were already there. @amaltaro what do you think?

**Comment by amaltaro** [comment]
If those 3 unit tests fail with the same reason as reported in this issue, then this would be an opportunity to have a PR fixing/closing it :-D Otherwise we just close it through the recently merged #9974 PR.

**Comment by mapellidario** [comment]
The test file `test/python/WMCore_t/WMSpec_t/WMWorkload_t.py` is related to slice 17 #10329. It' s already in the list of test files that I should check for slice 17, which is available [here](https://gitlab.cern.ch/cms-dmwm-test/wmcore-migration-status/-/blob/928f8c5fb421b9f1945ac9079df6cd8ac5786911/unittests/10141-17_t.txt), and as you can see in #10422 there is some work we have to do before getting to slice 17 :sweat_smile: 

So, if this is not a problem for you, in order not to open an issue for every failing unittest, I would stick to my schedule, postpone the work on this remaining 3 failing unittests, and close this issue because the specific unittest that was mentioned here has been fixed.

**Comment by mapellidario** [comment]
Anyway, two of the failing unittests are related to the usual `assertItemsEquals` -> `assertCountEqual`, while one is related to pickle and strings (ugh :sob: ), problems for future us!

```
[dmwm@70c3f941ccaf WMCore]$ nosetests test/python/WMCore_t/WMSpec_t/WMWorkload_t.py
....E........E.........E.........
======================================================================
ERROR: test persistency
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/dmwm/wmcore_unittest/WMCore/test/python/WMCore_t/WMSpec_t/WMWorkload_t.py", line 294, in testC
    workload.save(self.persistFile)
  File "/home/dmwm/wmcore_unittest/WMCore/src/python/WMCore/WMSpec/Persistency.py", line 49, in save
    pickle.dump(self.data, handle)
TypeError: write() argument must be str, not bytes

======================================================================
ERROR: Test Overrides setter and getter methods
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/dmwm/wmcore_unittest/WMCore/test/python/WMCore_t/WMSpec_t/WMWorkload_t.py", line 2112, in testOverrides
    self.assertItemsEqual(overrideArgs, result)
AttributeError: 'WMWorkloadTest' object has no attribute 'assertItemsEqual'

======================================================================
ERROR: _testUpdatingLFNAndDatasetStepChain_
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/dmwm/wmcore_unittest/WMCore/test/python/WMCore_t/WMSpec_t/WMWorkload_t.py", line 871, in testUpdatingLFNAndDatasetStepChain
    self.assertItemsEqual(testWorkload.getAcquisitionEra(), acqEra)
AttributeError: 'WMWorkloadTest' object has no attribute 'assertItemsEqual'

----------------------------------------------------------------------
Ran 33 tests in 0.169s

FAILED (errors=3)
[dmwm@70c3f941ccaf WMCore]$ 

```

**Comment by amaltaro** [comment]
Thanks for this explanation, Dario. Yes, I agree.
This issue has been fixed with #9974, thus closing it.
